pkg/validation/spec/responses.go

```diff
@@ -117,3 +127,59 @@ func (r *ResponsesProps) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+func (r *Responses) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Decoder) (err error) {
+	tok, err := dec.ReadToken()
+	if err != nil {
+		return err
+	}
+	var ext any
+	var resp Response
+	switch k := tok.Kind(); k {
+	case 'n':
+		return nil // noop
+	case '{':
+		for dec.PeekKind() != '}' {
```

Comment:
Since we always read the first token, we could this simplify as:
```go
for {
    tok, err := dec.ReadToken()
    if err != nil {
        return err
    }
    if tok.Kind() == '}' {
        return nil
    }
    ...
}
```